### PR TITLE
Display thumbnails in search results and show views

### PR DESCRIPTION
**Pull request by awead**
## Description

Using Blacklight's features and overriding them where needed, we're displaying thumbnails on search results and show views for works and file sets.

Connected to #740 

## Changes

* creates custom Blacklight presenters
* moves files storage to the public directory for easier linking
* indexes a thumbnail's path in the work and file set
* identifies representative file sets for works

### CODE REVIEW

File: app/cho/work/submission_indexer.rb [hunk 1]
```diff
@@ -21,7 +22,7 @@ def to_solr
       # @note returns the label associated with the work_type_id
       def label_for_id
         work_type = Work::Type.find(Valkyrie::ID.new(resource.work_type_id))
-        return resource.work_type_id if work_type.nil?
+        return resource.work_type_id.to_s if work_type.nil?
```
Comment (cam156) [inline app/cho/work/submission_indexer.rb]: @awead is this change needed?

File: app/views/catalog/_show_main_content.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<% @page_title = t('blacklight.search.show.title', document_title: document_show_html_title, application_name: application_name).html_safe %>
```
Comment (cam156) [inline app/views/catalog/_show_main_content.html.erb]: What is this override providing?

File: config/initializers/valkyrie.rb [hunk 3]
```diff
@@ -39,7 +39,7 @@
   # Storage Adapters
 
   Valkyrie::StorageAdapter.register(
-    Valkyrie::Storage::Disk.new(base_path: Rails.root.join('tmp', 'files')),
+    Valkyrie::Storage::Disk.new(base_path: Metrics::Repository.storage_directory),
```
Comment (cam156) [inline config/initializers/valkyrie.rb]: This seems odd having our initializer dependant on metrics.

File: spec/cho/work/submissions/indexer_spec.rb [hunk 4]
```diff
@@ -15,32 +15,30 @@
...
-      it { is_expected.to eq(work_type_ssim: "I don't exist", member_of_collection_ssim: nil) }
+      it { is_expected.to include(work_type_ssim: "I don't exist") }
     end
 
     context "when the resource's work type does exist" do
       let(:work_type) { create :work_type, label: 'Indexed Label' }
-      let(:resource) { instance_double('Resource', work_type_id: work_type.id) }
+      let(:resource)  { create(:work, work_type_id: work_type.id) }
 
-      it { is_expected.to eq(work_type_ssim: 'Indexed Label', member_of_collection_ssim: nil) }
```
Comment (cam156) [inline spec/cho/work/submissions/indexer_spec.rb]: Did member_of_collection_id go away for work?  Why is this needed for thumbnails?

File: app/cho/work/submission.rb [hunk 5]
```diff
@@ -34,8 +34,23 @@ def i18n_key
       end
     end
 
+    def file_sets
+      @file_sets ||= file_set_ids.map { |id| Work::FileSet.find(id) }
```
Comment (cam156) [inline app/cho/work/submission.rb]: I think this could get stale.  Aka if file_set_ids gets set on the work the @file_sets should be set to nil

File: app/cho/work/submission_indexer.rb (hunk 1 above)
Comment (awead) [reply app/cho/work/submission_indexer.rb]: It was a hidden bug that was exposed when I refactored the spec test. If the work's type has an id that doesn't exist yet, you'll get a solr field like: `work_type_ssim => #<Valkyrie::ID:0x007fdaa6852ef0 @id="I don't exist">` so we have to `to_s`. That being said, it doesn't make much sense for `work_type_ssim` to be an id if the type hasn't been saved yet or is non-existent. How about just return nil if the type isn't there:
``` ruby
return if work_type.nil?
```

File: app/views/catalog/_show_main_content.html.erb (hunk 2 above)
Comment (awead) [reply app/views/catalog/_show_main_content.html.erb]: Not sure. @mtribone ?

File: app/cho/work/submission.rb (hunk 5 above)
Comment (awead) [reply app/cho/work/submission.rb]: @cam156 Do you think not memoizing could have performance impacts if we need to re-query for files sets? Or, should be have a reload mechanism that can refresh these? Note, this applies in other areas too and not just file sets, so there's an opportunity to set a pattern here.

File: config/initializers/valkyrie.rb (hunk 3 above)
Comment (awead) [reply config/initializers/valkyrie.rb]: @cam156 Yeah, agreed. My intent is to refactor that out in another ticket. I didn't want it get too far afield. I can link a new ticket to this so we don't loose it.

File: spec/cho/work/submissions/indexer_spec.rb (hunk 4 above)
Comment (awead) [reply spec/cho/work/submissions/indexer_spec.rb]: @cam156 This was part to the refactor. I stopped using instance_double and started using the factory. This is also just testing work type, so I'm only verifying that work_type_ssim is set. In the factory, it already has a collection, so it's not nil anymore, but just checking for work type keeps the test more focused.

File: app/views/catalog/_show_main_content.html.erb (hunk 2 above)
Comment (mtribone) [reply app/views/catalog/_show_main_content.html.erb]: Ah! Not needed anymore as the thumbs and layout have been moved to another partial. Thanks!

Comment (mtribone) [reply app/views/catalog/_show_main_content.html.erb]: oh wait. It is to remove the pagination from the top when we are viewing works and filesets. It is needed.

Comment (mtribone) [reply app/views/catalog/_show_main_content.html.erb]: Nope. This feature was specifically asked for in the user stories. Putting it back in, but in another partial.

https://github.com/psu-libraries/cho/issues/124

File: config/initializers/valkyrie.rb (hunk 3 above)
Comment (awead) [reply config/initializers/valkyrie.rb]: Here's the ticket: #750

File: app/cho/work/submission_indexer.rb (hunk 1 above)
Comment (awead) [reply app/cho/work/submission_indexer.rb]: This was resolved in a previous commit.
